src/utils/entity-importers/index.js

```diff
@@ -0,0 +1,87 @@
+import Entities from 'constants/entities';
+
+import { has, pick, pickBy, mapValues } from 'constants/lodash';
+import { createId } from 'utils/common';
+
+const createMultipleImporter = (entitiyType, importFunction) => params => {
+  const { data, oldParent, parentEntity } = params;
+  if (!oldParent || !parentEntity) {
+    throw new Error(`Parent must be defined to import multiple ${entitiyType}`);
+  }
+  if (!has(data, entitiyType)) {
+    throw new Error(
+      `The import data has not right structure to import multiple ${entitiyType}`,
+    );
+  }
+
+  const keyMpping = Object.keys(
+    pickBy(
+      data[entitiyType],
+      entity =>
+        entity.parent === oldParent && entity.parentEntity === parentEntity,
+    ),
+  ).map(key => ({ oldKey: key, newKey: createId() }));
+  const children = keyMpping.reduce(
+    (obj, { oldKey, newKey }) => ({
+      ...obj,
+      [newKey]: importFunction({ ...params, key: oldKey }),
+    }),
+    {},
+  );
+  return { children, keyMpping };
+};
+
+const createImporter = (entitiyType, propertiesToPick) => ({
+  data,
+  key,
+  sector,
+  newParent,
+  parentEntity,
+}) => {
+  if (entitiyType !== Entities.sector.key && !sector) {
+    throw new Error(`Sector id must be defined to import a ${entitiyType}`);
+  }
+  if (!has(data, [entitiyType, key])) {
+    throw new Error(
+      `The import data has not right structure to import ${entitiyType}`,
```

Comment:
Same down here.

Reply:
One option would be to construct an object of translated strings to pass to these functions. The functions can just take and use the text they want.